internal/utils/utils.go

```diff
@@ -0,0 +1,48 @@
+package utils
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+)
+
+func GetNextOCPMinorVersion(versionString string) (*semver.Version, error) {
+	v, err := semver.Parse(versionString)
+	if err != nil {
+		return &v, err
+	}
+	v.Build = nil                 // Builds are irrelevant
+	v.Pre = nil                   // Next Y release
+	return &v, v.IncrementMinor() // Sets Y=Y+1 and Z=0
+}
+
+func ToAllowedSemver(value string) (*semver.Version, error) {
+	if value == "" {
```

Comment:
This doesn't look like it'll work for this scenario:
```go
value := `"4.16"`
vers, err := ToAllowedSemver(value)
```

To fix, I think I'd suggest parsing as JSON, something like the following. I don't think it's a good idea to use naive prefix and suffix trimming of double quotes because there are likely JSON complexities that that method would gloss over.

```go
var rawV interface{}
if err := json.Unmarshal(&rawV, value); err != nil {
      return nil, err
}
switch val := rawV.(type) {
case string:
    // handle string
case <whateverTheRightFloatTypeIs>:
    // handle float, but need to think about weird rounding that might happen
    // maybe we see that it's a float, but then parse the raw value like you already are?
default:
      return nil, fmt.Errorf("invalid type %T for olm.maxOpenshiftVersion: %s", string(value))
}
```

Probably makes sense to add some unit tests here to make sure we have our bases covered.